jenkins/cleanup_python_code.py

```diff
@@ -11,8 +11,7 @@
 @click.command()
 @click.option(
     '--sha',
-    help="Sha of the merge commit to base the new PR off of",
```

Comment:
Are there situations where this might not be the current SHA?

Reply:
As far as I can tell, nope! Here are the only data sources I could find for the value of these arguments, both calling `git rev-parse HEAD`:

- https://github.com/edx/jenkins-job-dsl/blob/63fd1469303474ff5985f2ee73c2c16ddfc8572b/testeng/resources/cleanup-python-code-create-pr.sh#L28
- https://github.com/edx/jenkins-job-dsl/blob/909a4840fedb417637dee52bf611908294d6a28c/testeng/resources/upgrade-python-requirements.sh#L13

And here are the two places it is used:

- [Making a deterministic branch name](https://github.com/edx/testeng-ci/blob/8cc13f370a51c472a865d5cd1cb2ce235492936d/jenkins/pull_request_creator.py#L59)
- [Building a new commit](https://github.com/edx/testeng-ci/blob/8cc13f370a51c472a865d5cd1cb2ce235492936d/jenkins/github_helpers.py#L210), which could conceivably take a SHA that is different from the current (presumably dirty) checkout

It's never actually used to perform a checkout operation, so there's an implicit assumption that it's the same as the current commit.

Reply:
It feels like we should either:
1. Compare this input to the sha you are finding and raise an exception if they are different with a message that states that that this option should no longer be used, and/or
2. Clarify that you have a plan to quickly follow-up with removing use of this option, and then deleting it, or
3. Other?

Reply:
Option 2 sounds good. I can ticket up the work for removing deprecated options, for after this work gets merged.

Reply:
Filed ARCHBOM-1632 for followup.